avalon/tools/creator/app.py

```diff
@@ -20,6 +21,91 @@
 Separator = "---separator---"
 
 
+class SubsetNameValidator(QtGui.QRegExpValidator):
```

Comment:
To be sure, the only reason we require to define our own class is so we can report to the user which characters were invalidated in particular. Correct? If so, then this class is fine. Otherwise, this seems redundant.

Reply:
> Correct?

Not entirely, the main reason I subclass the `QtGui.QRegExpValidator` was to override the function `validate`, so that I could emit a signal when the input has invalid character. I could not find any built-in way to implement this notification on `QLineEdit`, it just works silently.

Reporting which characters exactly were invalid was something I picked up on the way :)

Reply:
Ah. If you only wanted to have the invalid signal, it's [`QLineEdit.inputRejected`](https://doc.qt.io/qt-5/qlineedit.html#inputRejected) but it won't signal what characters were invalid I believe + it's only been [introduced since Qt 5.12](https://code.qt.io/cgit/qt/qtbase.git/tree/dist/changes-5.12.0/?h=v5.12.0#n349) so I guess we can't rely on it.